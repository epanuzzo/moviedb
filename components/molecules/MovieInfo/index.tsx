import React from "react";

export type MovieInfoProps = {
  name: string;
  children: React.ReactNode;
};

/**
 * A React functional component that displays movie information.
 *
 * @component
 * @param {MovieInfoProps} props - The properties object.
 * @param {string} props.name - The name of the movie.
 * @param {React.ReactNode} props.children - The additional information about the movie.
 * @returns {JSX.Element} A paragraph element containing the movie name and additional information.
 */
const MovieInfo: React.FC<MovieInfoProps> = ({ name, children }) => {
  return (
    <p>
      <span className="font-semibold">{name}:</span> {children}
    </p>
  );
};

export default MovieInfo;
