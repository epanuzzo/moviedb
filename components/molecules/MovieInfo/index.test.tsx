import React from "react";
import renderer from "react-test-renderer";
import MovieInfo from "./index";

describe("MovieInfo", () => {
  it("renders component", () => {
    const tree = renderer
      .create(<MovieInfo name="Name">Child</MovieInfo>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
