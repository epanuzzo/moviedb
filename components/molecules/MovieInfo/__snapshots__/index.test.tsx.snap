// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieInfo renders component 1`] = `
<p>
  <span
    className="font-semibold"
  >
    Name
    :
  </span>
   
  Child
</p>
`;
